src/Symfony/Component/Notifier/Bridge/Mercure/Tests/MercureTransportTest.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Mercure\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Mercure\PublisherInterface;
+use Symfony\Component\Mercure\Update;
+use Symfony\Component\Notifier\Bridge\Mercure\MercureOptions;
+use Symfony\Component\Notifier\Bridge\Mercure\MercureTransport;
+use Symfony\Component\Notifier\Exception\InvalidArgumentException;
+use Symfony\Component\Notifier\Exception\LogicException;
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Message\ChatMessage;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\MessageOptionsInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+use TypeError;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ */
+final class MercureTransportTest extends TestCase
```

Comment:
As Mercure transport is a bit special (it doesn't deal with real HTTP Dsn, and it use a publisher instead of an HTTP client), I added extra tests and marked `testCanSetCustomPort`, `testCanSetCustomHost` and `testCanSetCustomHostAndPort` as skipped.